cli/tests/pcluster/config/test_config_patch.py

```diff
@@ -0,0 +1,315 @@
+                    UpdatePolicy.UNSUPPORTED,
+                    fail_reason="EBS sections cannot be added or removed during a 'pcluster update' operation",
+                ),
+            ),
+            Change("ebs", "ebs-1", "shared_dir", None, "vol1", UpdatePolicy(UpdatePolicy.UNSUPPORTED)),
+        ],
+        UpdatePolicy.UNSUPPORTED,
+    )
+
+
+def _test_incompatible_ebs_sections(base_conf, target_conf):
+    # Change shared_dir param value in target conf
+    target_conf.get_section("ebs", "ebs-1").get_param("shared_dir").value = "new_value"
+
+    # The patch must show the updated shared_dir for ebs-1 section
+    _check_patch(
+        base_conf,
+        target_conf,
+        [Change("ebs", "ebs-1", "shared_dir", "vol1", "new_value", UpdatePolicy(UpdatePolicy.UNSUPPORTED))],
+        UpdatePolicy.UNSUPPORTED,
+    )
+
+
+def _test_different_labels(base_conf, target_conf):
+    # First make sure sections are present with original labels
+
+    base_ebs_1_section = base_conf.get_section("ebs", "ebs-1")
+    base_ebs_2_section = base_conf.get_section("ebs", "ebs-2")
+
+    assert_that(base_conf.get_section("ebs", "ebs-1")).is_not_none()
+    assert_that(base_conf.get_section("ebs", "ebs-2")).is_not_none()
+
+    # Now update section labels and make sure they're not more present with original labels
+    base_ebs_1_section.label = "ebs-1_updated"
+    base_ebs_2_section.label = "ebs-2_updated"
+    base_conf.refresh()
+
+    assert_that(base_conf.get_section("ebs", "ebs-1_updated")).is_not_none()
+    assert_that(base_conf.get_section("ebs", "ebs-2_updated")).is_not_none()
+    assert_that(base_conf.get_section("ebs", "ebs-1")).is_none()
+    assert_that(base_conf.get_section("ebs", "ebs-2")).is_none()
+
+    # The patch should contain 5 differences:
+    # - 2 volumes in target conf not matched in base conf
+    # - 2 volumes in base conf not matched in target conf
+    # - 1 ebs_settings changed in cluster section
+    _check_patch(
+        base_conf,
+        target_conf,
+        [
+            Change("ebs", "ebs-1", "shared_dir", None, "vol1", UpdatePolicy(UpdatePolicy.UNSUPPORTED)),
+            Change("ebs", "ebs-1_updated", "shared_dir", "vol1", None, UpdatePolicy(UpdatePolicy.UNSUPPORTED)),
+            Change("ebs", "ebs-2", "shared_dir", None, "vol2", UpdatePolicy(UpdatePolicy.UNSUPPORTED)),
+            Change("ebs", "ebs-2_updated", "shared_dir", "vol2", None, UpdatePolicy(UpdatePolicy.UNSUPPORTED)),
+            Change(
+                "cluster",
+                "default",
+                "ebs_settings",
+                "ebs-1_updated,ebs-2_updated",
+                "ebs-1,ebs-2",
+                UpdatePolicy(
+                    UpdatePolicy.UNSUPPORTED,
+                    fail_reason="EBS sections cannot be added or removed during a 'pcluster update' operation",
+                ),
+            ),
+        ],
+        UpdatePolicy.UNSUPPORTED,
+    )
+
+
+@pytest.mark.parametrize(
+    "test",
```

Comment:
Why do you think it's harder to follow the tests? The output is clear since you can directly see which test is being done:
```
test_config_patch.py::test_adaptation[_test_less_target_sections] PASSED [ 69%]
test_config_patch.py::test_adaptation[_test_more_target_sections] PASSED [ 76%]
test_config_patch.py::test_adaptation[_test_incompatible_ebs_sections] PASSED [ 84%]
test_config_patch.py::test_adaptation[_test_equal_configs] PASSED        [ 92%]
test_config_patch.py::test_adaptation[_test_different_labels] PASSED     [100%]
```